runtime/interpreter/interpreter.go

```diff
@@ -1180,30 +1154,23 @@ func (interpreter *Interpreter) visitAssignment(
 	valueExpression ast.Expression, valueType sema.Type,
 	position ast.HasPosition,
 ) {
-
 	// First evaluate the target, which results in a getter/setter function pair
 	getterSetter := interpreter.assignmentGetterSetter(targetExpression)
 
 	getLocationRange := locationRangeGetter(interpreter.Location, position)
 
-	if targetMemberExpression, ok := targetExpression.(*ast.MemberExpression); ok {
-		interpreter.checkMemberAccessedType(
-			targetMemberExpression,
-			getterSetter.target,
-			getLocationRange,
-		)
-	}
```

Comment:
We have the container mutation check in place now, so this is temporary solution we had added initially is not necessary anymore